cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/oci_repo.go

```diff
@@ -526,29 +462,56 @@ func (s *repoEventSink) onModifyOciRepo(key string, oldValue interface{}, repo s
 			key, cacheEntryUntyped)
 	}
 
-	ociRepo, err := s.newOCIChartRepositoryAndLogin(context.Background(), repo)
+	ociChartRepo, err := s.newOCIChartRepositoryAndLogin(context.Background(), repo)
 	if err != nil {
 		return nil, false, err
 	}
 
-	newChecksum, err := ociRepo.checksum()
+	appNames, err := ociChartRepo.listRepositoryNames()
+	if err != nil {
+		return nil, false, err
+	}
+
+	allTags, err := ociChartRepo.getTagsForApps(appNames)
+	if err != nil {
+		return nil, false, err
+	}
+
+	newChecksum, err := ociChartRepo.checksum(appNames, allTags)
 	if err != nil {
 		return nil, false, err
 	}
 
 	if cacheEntry.Checksum != newChecksum {
-		// currently this cannot happen due to https://github.com/fluxcd/source-controller/issues/839
-		// skip because the content did not change
-		log.Warningf("Unexpected state in onModifyOciRepo(%s)", common.PrettyPrint(repo))
-		// then just move on, same as if the remote contents did not change
+		charts, err := getOciChartModels(appNames, allTags, ociChartRepo, &repo)
+		if err != nil {
+			return nil, false, err
+		}
+
+		cacheEntryValue := repoCacheEntryValue{
+			Checksum: newChecksum,
+			Charts:   charts,
+			Type:     "oci",
+		}
+
+		// use gob encoding instead of json, it peforms much better
+		var buf bytes.Buffer
+		enc := gob.NewEncoder(&buf)
+		if err = enc.Encode(cacheEntryValue); err != nil {
+			return nil, false, err
+		}
+
+		if s.chartCache != nil {
+			fn := downloadOCIChartFn(ociChartRepo)
+			if err = s.chartCache.SyncCharts(charts, fn); err != nil {
```

Comment:
I guess this is still syncing by just adding any new versions, no? I mean, in the event of a chart deprecation, the package won't get removed from the cache, no?
If so, wouldn't the upstream checksum always differ from the cached one which may lead to a permanent cache-miss state? 

Not sure, just asking, though.

Reply:
Ah, ok, ok, got it. If we aren't looking at the charts in the local cache for computing it, then it doesn't seem an issue. Thanks for the clarification!